### ISSUE TITLE
Add pinball loss for quantile regression

### ISSUE THREAD

**Issue by georgedahl** [google/jaxopt#408]
It would be really useful to have the pinball loss defined. Similar to sklearn https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_pinball_loss.html
